src/components/organisms/SchedulePageModal/SchedulePageModal.tsx

```diff
@@ -1,162 +1,126 @@
-    const hasEvents = liveAndFutureEvents.length > 0;
-
-    const nowDay = startOfDay(new Date());
-
-    const dates: DatedEvents = range(0, DAYS_AHEAD).map((idx) => {
-      const day = addDays(nowDay, idx);
-
-      const todaysEvents = liveAndFutureEvents
-        .filter((event) => {
-          return isWithinInterval(day, {
-            start: startOfDay(new Date(event.start_utc_seconds * 1000)),
-            end: endOfDay(
-              new Date(
-                (event.start_utc_seconds + event.duration_minutes * 60) * 1000
-              )
-            ),
-          });
-        })
-        .sort((a, b) => a.start_utc_seconds - b.start_utc_seconds);
-
-      return {
-        dateDay: day,
-        events: hasEvents ? todaysEvents : [],
-      };
-    });
-
-    return dates;
-  }, [relatedVenueEvents]);
-
-  const [date, setDate] = useState(0);
-
-  const scheduleTabs = useMemo(
+    const today = startOfDay(Date.now());
+    const buildScheduleEvent = scheduleDayBuilder(
+      today,
+      liveAndFutureEvents,
+      relatedRooms,
+      userEventIds
+    );
+
+    return range(0, DAYS_AHEAD).map((dayIndex) => buildScheduleEvent(dayIndex));
+  }, [relatedVenueEvents, relatedRooms, userEventIds]);
+
+  const [selectedDayIndex, setSelectedDayIndex] = useState(0);
+
+  const onWeekdayClick = useCallback(
+    (index: number): MouseEventHandler<HTMLLIElement> => (e) => {
+      e.stopPropagation();
+      setSelectedDayIndex(index);
+    },
+    []
+  );
+
+  const weekdayClasses = useCallback(
+    (index) =>
+      classNames("SchedulePageModal__weekday", {
+        "SchedulePageModal__weekday--active": index === selectedDayIndex,
+      }),
+    [selectedDayIndex]
+  );
+
+  const weekdays = useMemo(
     () =>
-      orderedEvents.map((day, idx) => (
+      schedule.map((day, index) => (
         <li
-          key={formatDate(day.dateDay.getTime())}
-          className={`button ${idx === date ? "active" : ""}`}
-          onClick={(e) => {
-            e.stopPropagation();
-            setDate(idx);
-          }}
+          key={formatDate(day.dayStartUtcSeconds)}
+          className={weekdayClasses(index)}
+          onClick={onWeekdayClick(index)}
         >
-          {formatDateToWeekday(day.dateDay.getTime() / 1000)}
+          {day.isToday ? "Today" : day.weekday}
         </li>
       )),
-    [date, orderedEvents]
+    [schedule, weekdayClasses, onWeekdayClick]
   );
 
-  const events = useMemo(
+  const containerClasses = useMemo(
     () =>
-      orderedEvents[date]?.events.map((event, index) => (
-        <EventDisplay
-          key={event.id ?? `${index}-${event.name}`}
-          event={event}
-          venue={relatedVenuesById[event.venueId] ?? currentVenue}
-        />
-      )),
-    [currentVenue, date, orderedEvents, relatedVenuesById]
+      classNames("SchedulePageModal", { "SchedulePageModal--show": isVisible }),
+    [isVisible]
   );
```

Comment:
We don't need to `useMemo` the result of `classNames` as it's just a plain `string`, and super quick/lightweight to run.

Reply:
removed that: https://github.com/sparkletown/sparkle/pull/1302/commits/6c79ed94ad0acdbf8061bba26c4e73f7912ed2de